File: JPA/board-api/src/main/java/me/kzv/boardapi/persistence/UserRepository.java
package me.kzv.boardapi.persistence;

import me.kzv.boardapi.domain.User;
import me.kzv.boardapi.dto.SingleResultDto;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
}

File: hello-kotlin/kotlin-board/src/main/kotlin/me/kzv/board/repository/UserRepository.kt
package me.kzv.board.repository

import me.kzv.board.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserRepository : JpaRepository<User, Long> {
    fun findByUserIdAndPassword(userId: String, password: String): Optional<User>
}